packages/rpc-core/src/rpc-methods/getSignaturesForAddress.ts

```diff
@@ -0,0 +1,44 @@
+import { Base58EncodedAddress } from '@solana/keys';
+import { Commitment, RpcResponse, Slot, TransactionSignature, U64UnsafeBeyond2Pow53Minus1 } from './common';
+
+type GetSignaturesForAddressTransaction = RpcResponse<{
+    /** transaction signature as base-58 encoded string */
+    signature: TransactionSignature;
+    /** The slot that contains the block with the transaction */
+    slot: Slot;
+    /** Error if transaction failed, null if transaction succeeded. */
+    err: object | null;
+    /** Memo associated with the transaction, null if no memo is present */
+    memo: string | null;
+    /** estimated production time, as Unix timestamp (seconds since the Unix epoch) of when transaction was processed. null if not available. */
+    blockTime: U64UnsafeBeyond2Pow53Minus1 | null;
+    /** The transaction's cluster confirmation status */
+    confirmationStatus: Commitment | null;
+}>;
+
+type GetSignaturesForAddressApiResponse = [GetSignaturesForAddressTransaction];
+
+export type AllowedCommitmentForGetSignaturesForAddress = Exclude<Commitment, 'processed'>;
```

Comment:
<3

Reply:
I'd err on the side of not exporting stuff. I see all you've needed this for is a test. You can do this instead of having to export it.

```ts
['confirmed', 'finalized'] as Parameters<typeof rpc.getSignaturesForAddress>[1]['commitment'][]
```

Reply:
Nice, makes sense! TypeScript doesn't seem to like it though ([playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAggNnA9gdwgEwMKILbYJbDYQB2wAYogE4DiEwAyngObECGwArpRAM4WUw0abjx5QAvFADkAY0TEAZnkpE0UqAB9pStnDwAvdFIDcAKFOhIUWg2ZtOI-oOG8eWRcwlQAShFZp5OBAAHgBvUyhIqDlcAiJSAH4ALlgEFHQsWMISciobRhZ2Ll4nIREeMyioAHoAKlqoABUAC2h8YjxsDmwoHiRgKGBm9kHWqG4ARw5eAZlWYigAI2gIADdWOA52dCgR2uqIqPb3YAgADwZ+5KhibuXKSqi6huxWM87uwcp5nlYZYDw8l6diKIkGiHGdC4CwAFMtgKgSFAAIy7YhoFEAGgADLiAJQAOigABEIApWBw4MAUsjcdioPtDpE9PhgNdbth7o9Is9esBWJQBjw-JQZM08MQmEs-gBrZACtBiBSUHCjPBiYDfYi-f6AhY8EEOCBEgCSChuiAGYBVqzwaB2Q2gwoFYr5AuASpVPUd4LAUEQ5p94qYrR4A1eZ2i8iUKh2iyQMhlRMZVWWCioEGuYcoEqY3Jq9V6ItdHFIeDgao1Wp1AKBBsKRsxUDw5vTpYxaYzUBZBEhf1aGJTUVLALgWc1ubMAF8AHxmUwKUu6oFMOgFezFPhUZzlGGHfwuUQpbO5zGHOQeJjXfKGzelQ9uaPMUx4lKIRYAKwg-yg4Ui3AcBZQinadzEsaBMlZeJgBNYgwA4YBGnAaBJAABQFVgiFOSgeGCcCAygVdbAbO9tzKVwZwAbWRABdSjZBwKCcikGjjCAA))

Property 'commitment' does not exist on type 'Readonly<{ commitment?: AllowedCommitmentForGetSignaturesForAddress | undefined; minContextSlot?: number | undefined; limit?: number | undefined; before?: string | undefined; until?: string | undefined; }> | undefined'.(2339)

I think it's because the `config` object is optional, do you know if there's a way to unwrap that in this type definition?

Reply:
Yeah! Add `NonNullable` ([playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAggNnA9gdwgEwMKILbYJbDYQB2wAYogE4DiEwAyngObECGwArpRAM4WUw0abjx5QAvFADkAY0TEAZnkpE0UqAB9pStnDwAvdFIDcAKFOhIUWg2ZtOI-oOG8eWRcwlQAShFZp5OBAAHgBvUyhIqDlcAiJSAH4ALlgEFHQsWMISciobRhZ2Ll4nIREeMyioAHoAKlqoABUAC2h8YjxsDmwoHiRgKGBm9kHWqG4ARw5eAZlWYigAI2gIADdWOA52dCgR2uqIqPb3YAgADwZ+5KhibuXKSqi6huxWM87uwcp5nlYZYDw8l6diKIkGiHGdC4CwAFMtgKgSFAAIy7YhoFEAGgADLiAJQAOigABEIApWBw4MAUsjcdioPtDpE9PhgNdbth7o9Is9esBWJQBjw-JQZM08MQmEs-gBrZACtBiBSUHCjPBiYDfYi-f6AhY8EEOCBEgCSChuiAGYBVqzwaB2Q2gwoFYr5AuASpVPUd4LAUEQ5p94qYrR4A1eZ2i8iUKh2iyQMhlRMZVWWCioEGuYcoEqY3Jq9V6ItdHFIeDgao1Wp1AKBBsKRsxUDw5vTpYxaYzUBZBEhf1aGJTUVLALgWc1ubMAF8AHxmUwKUu6oFMOgFezFPhUZzlGGHfwuUQpbO5zGHOQeJjXfKGzelQ9uaPMUx4lKIRYAKwg-yg4Ui3AcBZQinadzEsaBMlZeJgBNYgwA4YBGnAaBJAAOXkVDKTgVh4wgYIAAUBVYIhTkoHhgnAgMoFXWwGzvbcylcGcAG1kQAXRY2QcCgnIpDY4wgA)).